core/domain/exp_jobs_one_off.py

```diff
@@ -393,16 +403,29 @@ def map(item):
                 'to_version': python_utils.UNICODE(
                     feconf.CURRENT_STATE_SCHEMA_VERSION)
             })]
-            exp_services.update_exploration(
-                feconf.MIGRATION_BOT_USERNAME, item.id, commit_cmds,
-                'Update exploration states from schema version %d to %d.' % (
-                    item.states_schema_version,
-                    feconf.CURRENT_STATE_SCHEMA_VERSION))
+            try:
+                exp_services.update_exploration(
+                    feconf.MIGRATION_BOT_USERNAME, item.id, commit_cmds,
+                    'Update exploration states from schema version %d to %d.' %
+                    (
+                        item.states_schema_version,
+                        feconf.CURRENT_STATE_SCHEMA_VERSION))
+            except Exception as e:
```

Comment:
Just to check -- there's no chance that an exception is thrown _after_ the exploration is successfully updated, right?

Reply:
Hmm, I can't think of any issue. The only step after this is the return of job output, right? 
If the saving of explorations to datastore is asynchronous (which I'm not sure of), then there can be an issue there, but even that should be caught here since this is the last function in the job and we yield just after this.

Reply:
OK, sg!